Cabal/Distribution/PackageDescription/Check.hs

```diff
@@ -253,6 +253,18 @@ checkLibrary pkg lib =
            "An 'autogen-module' is neither on 'exposed-modules' or "
         ++ "'other-modules'."
 
+    -- check that all autogen-c-sources appear on c-sources
+  , check
+      (not $ and $ map (flip elem
+        (cSources $ libBuildInfo lib)) (autogenCSources $ libBuildInfo lib)) $
+      PackageBuildImpossible $ "An 'autogen-c-sources' is not on 'c-sources'"
```

Comment:
This error would be better if (1) it explained WHY you had to duplicate them (it's for backwards compatibility with old Cabal) and (2) said that it was sufficient to just copy paste.